### PR TITLE
Added support for Swarm Service Driller (ssd)

### PR DESCRIPTION
**Pull request by quadespresso**

### CODE REVIEW

**Comment by quadespresso** [comment]
Ref: https://github.com/docker/orca/issues/6793

**Comment by quadespresso** [comment]
Ping @fcrisciani @ctelfer @ddebroy PTAL

**Comment by quadespresso** [comment]
@adamancini yes: https://github.com/docker/orca/blob/master/images/dsinfo/Dockerfile#L36

Also: still waiting on this to be merged, which will address where `ssd.py` gets pulled from, once merged: https://github.com/docker/orca/pull/12210

**Comment by trapier** [comment]
Am focusing on other work activities through at least next Tuesday March 27.  Please nominate another reviewer if this needs attention before then.

File: support.sh [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+SSD="${SSD:-false}"
+
 # Required tools
 DOCKER="${DOCKER:-docker}"
 NSENTER="${NSENTER:-nsenter}"
 BRIDGE="${BRIDGE:-bridge}"
 IPTABLES="${IPTABLES:-iptables}"
 IPVSADM="${IPVSADM:-ipvsadm}"
 IP="${IP:-ip}"
+SSDBIN="${SSDBIN:-/ssd}"
```
Comment (adamancini) [inline support.sh]: @quadespresso if I drop a `ssd` binary (or shell script calling 'docker run ... ssd') in my $PATH to test, the leading `/` breaks the lookup for the command - I think we can we depend on $PATH inside the container, can we change this to `SSDBIN="${SSDBIN:-ssd}" `

Comment (adamancini) [reply support.sh]: https://gist.github.com/adamancini/116ae89b260dc64b67ee46e86ef3a9b1 are my results from support.sh when I call it with `ssd` support

Comment (adamancini) [reply support.sh]: LGTM otherwise

Comment (quadespresso) [reply support.sh]: As discussed on Slack with @adamancini, I'll make the change as discussed above to remove the leading `/`.

Likewise, I'll update `Dockerfile` in `dsinfo` to `/bin` in order to leverage `$PATH` correctly (and to be consistent with a couple other items we're copying to that directory).

File: support.sh [hunk 2]
```diff
@@ -102,3 +113,17 @@ echo -e "\n\n==SUMMARY=="
 echo -e "\t Processed $networks networks"
 echo -e "\t IP overlap found: $ip_overlap"
 echo -e "\t Processed $containers containers"
+
+if [ "true" == ${SSD} ] ; then
```
Comment (fcrisciani) [inline support.sh]: not sure if the SSD has some output, but I would like the SUMMARY to be the last thing to be printed at the end

Comment (quadespresso) [reply support.sh]: ${SSD} is just something getting passed from another script to enable the ssd code. If I've understood your question correctly?

Comment (fcrisciani) [reply support.sh]: I put the comment on the top of the block but was referring to the underlying code. I would like to keep the summary as the last output produced from the script, so you need to move your change on top

File: support/support.sh [hunk 3]
```diff
@@ -101,6 +112,20 @@ while read containerID status; do
...
 done < <(${DOCKER} container ls -a --format '{{.ID}} {{.Status}}' |cut -d' ' -f1,2)
 
+if [ "true" == ${SSD} ] ; then
+    echo "" ; echo "#### SSD control-plane and datapath consistency check on a node ####"
+    for netName in $(docker network ls -f driver=overlay --format "{{.Name}}") ; do
+        echo "## $netName ##"
+        ${SSDBIN} $netName
+        echo ""
+    done
+    echo "" ; echo "#### SSD control-plane consistency check across nodes in a cluster ####"
```
Comment (dani-docker) [inline support/support.sh]: @quadespresso let's take `gossip-consistency` out for now, we'll add it back once we address the caller

**Comment by ctelfer** [comment]
Looks like all comments were addressed.   The unit tests that failed clearly had nothing to do with this change as it touches none of the go code, but the tests that were failing were go tests.

**Pull request merged**